numpy/ma/core.py

```diff
@@ -1613,7 +1613,11 @@ def make_mask(m, copy=False, shrink=True, dtype=MaskType):
             else:
                 result = m
         else:
-            result = np.array(m, dtype=dtype, copy=copy)
+            # existence of fields implies True
+            if m.dtype.fields and np.dtype(dtype) == np.bool_:
+                result = np.ones(m.shape, dtype=dtype)
```

Comment:
It does what you'd expect if you pass `np.ma.make_mask(a, dtype='b,b')` on 1.11, but fails even then on master.